### ISSUE TITLE
Do variant descriptions containing alleles AND intronic positions break VV?

### ISSUE THREAD

**Issue by leicray** [openvar/variantValidator#181]
The description NC_000017.11(NM_000154.1):c.[700del];[(?_945-7)_(*64del_?)] describes allelic variants in the _GALK1_ gene leading galactokinase 1 deficiency. The first variant is a straightforward single-nucleotide deletion detected by sequencing. The second variant is a deletion with unknown endpoints detected by qPCR that deletes all or most of the final two exons of the gene.

Because one endpoint in the second variant lies in an intron, the variant cannot be validated in the context of a transcript reference sequence alone. I tried to validate the basic components of the sequence description by first removing the uncertain endpoints. This yields:
`NC_000017.11(NM_000154.1):c.[700del];[945-7_*64del_?]`
but it generates an error message that suggests that VV is not parsing the input string correctly:

```
Error
Unable to validate the submitted variant NC_000017.11(NM_000154.1):c.[700del];[*64del] against
the GRCh38 assembly. The following warnings were returned:

NM_000154.1:c.[700del];[*64del]: char 14: expected one of (, *, or a digit
Please check your submission and re-submit.
```
If I disassemble the description into smaller component parts, each part validates correctly.

I cannot see any syntax errors in the input and wonder if the problem is that VV is failing to correctly parse the input string.

**Comment by PeteCausey-Freeman** [comment]
Highly possible. This format is again another total mess for computational parsing. I'll take a look

**Comment by PeteCausey-Freeman** [comment]
It is also likely a product of several other processing steps

Note, NC_000017.11(NM_000154.1):c.[700del];[945-7_*64del_?]. We cannot support uncertain positions. I'll come up with something.

**Comment by PeteCausey-Freeman** [comment]
Also, what is the meaning of del_?  ?

**Comment by leicray** [comment]
Thank goodness for that. I was expecting you to immediately point out a flaw in my assembly of the the variant description. I have checked it against the recommendations for alleles and for uncertain endpoints and think that the syntax is correct. I did strip out the uncertain positions to give VV a fighting chance, but something is tripping it up.

**Comment by leicray** [comment]
Quite right!

It should be NC_000017.11(NM_000154.1):c.[700del];[945-7_*64del] of course.

But that still does not parse correctly.

**Comment by PeteCausey-Freeman** [comment]
Thanks,

I can see a couple of potential reasons. I did not test with the format NC_000017.11(NM_000154.1) and I cannot remember whether I coded up support for intronic positions. I'll take a look. This may well be one of those cases where if the reference sequences for the genomic is provided we do something different to it not being provided. Really nice case study

**Comment by ifokkema** [comment]
Sorry to interrupt here;

> `NC_000017.11(NM_000154.1):c.[700del];[(?_945-7)_(*64del_?)]`

I believe you mean `(?_945-7)_(*64_?)del` for the last variant.
Removing the uncertainties, you indeed get to the description `945-7_*64del`.

Then:

> It should be NC_000017.11(NM_000154.1):c.[700del];[945-7_*64del] of course.
> 
> But that still does not parse correctly.

In my opinion, it wouldn't be fair to send this to VV. These are multiple alleles, and therefore VV should run *twice* to validate these, and give you *two results*. I would recommend users to split this into `NC_000017.11(NM_000154.1):c.700del` and `NC_000017.11(NM_000154.1):c.945-7_*64del` and process the respective results.

That is separate from `NC_000017.11(NM_000154.1):c.[700del;945-7_*64del]` not validating either, which is something that can be looked into.

**Comment by PeteCausey-Freeman** [comment]
Thanks Ivo, All comments appreciated. I'll drop in my thoughts when I start debugging this for both of you to comment.

**Comment by PeteCausey-Freeman** [comment]
https://github.com/openvar/variantValidator/commits/features

**Comment by leicray** [comment]
Has this propagated to the live site yet?

I ask because NC_000017.11(NM_000154.1):c.[700del];[(?_945-7)_(*64_?)del] still breaks the validator.

**Comment by leicray** [comment]
Apologies, GitHub is interpreting the variant description in a strict markdown fashion. The variant is:

`NC_000017.11(NM_000154.1):c.[700del];[(?_945-7)_(*64_?)del]`

**Comment by PeteCausey-Freeman** [comment]
> I ask because NC_000017.11(NM_000154.1):c.[700del];[(?945-7)(*64_?)del] still breaks the validator.

No, I only just pushed up the code. I will update shortly. Fixing the code does not update the servers. 

I'll do it now though. Give it 15 min or so

**Comment by leicray** [comment]
Apologies for my impatience.

**Comment by PeteCausey-Freeman** [comment]
No problem. Should be done now. Please let me know if it works. Not had chance to test the core updates on the interface. You just volunteered for user acceptance testing :)

**Comment by leicray** [comment]
Validation now returns:

`Unsupported format c.[700del];[(?_945-7_(*64_?del]`

I know that that's correct in a formal sense, but a more explicit error message could say that variant descriptions containing uncertain nucleotide positions (?) are unsupported. I also wonder if it might be possible to return a message that confirms whether the syntax is correct. A user might wish to have that reassurance even if a full validation is not possible.

**Comment by ifokkema** [comment]
Funnily enough, sending only [`NC_000017.11(NM_000154.1):c.(?_945-7)_(*64_?)del`](https://www35.lamp.le.ac.uk/VariantValidator/variantvalidator/GRCh37/NC_000017.11(NM_000154.1)%3Ac.(%3F_945-7)_(*64_%3F)del/all?content-type=application%2Fjson) gives a more informative `Uncertain positions are not currently supported`.

**Comment by PeteCausey-Freeman** [comment]
NC_000017.11(NM_000154.1):c.[700del];[(?_945-7)_(*64_?)del] will not be captured correctly because I have not added code into the allele capture to deal with uncertain positions. 

This nomnenclature is a total mess sometimes :)
